### PR TITLE
Update circle.yml to deploy docs based on tags

### PR DESCRIPTION
**Pull request by theseanything**
Changed the circleci deployment to only update docs at /docs/ with a new tag version and /dev/doc/ with the latest from master branch.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -17,12 +17,10 @@ test:
     - make docs
 deployment:
     production:
-        branch: master
+        tag: /v*/
         commands:
-            - make docs
             - aws s3 sync ~/sceptre/docs/_build/html s3://sceptre.cloudreach.com/docs/ --delete
```
Comment (jamesroutley) [inline circle.yml]: @theseanything Sorry - I'd written this comment a while ago but hadn't submitted it. Still getting used to GH I guess.

Comment (theseanything) [inline circle.yml]: Ah do you mean - that only certain tagged commits get push to docs or make sure tagged commits ALSO get put on dev?

Comment (jamesroutley) [inline circle.yml]: Make sure tagged commits also get pushed to dev

Comment (theseanything) [inline circle.yml]: Yup so thats what would happen - only commits with `v*` tags would go to `/docs` and all commits regardless of tags goes to `/dev/docs`

Comment (jamesroutley) [inline circle.yml]: cool

**Pull request merged**